### PR TITLE
Add text-to-SQL loader

### PR DESCRIPTION
**Pull request by shuaichenchang**
# Overall
The second step of adding the text-to-SQL task in https://github.com/neulab/ExplainaBoard/pull/580 
# Details
Standard input files in the text-to-SQL task use a JSON file containing the database id, question, and gold query in each example.  Standard output files are text files, where each line has a predicted query.

### CODE REVIEW

File: explainaboard/loaders/loader_factory.py [hunk 1]
```diff
@@ -31,6 +31,7 @@
 from explainaboard.loaders.tabular_regression import TabularRegressionLoader
 from explainaboard.loaders.text_classification import TextClassificationLoader
 from explainaboard.loaders.text_pair_classification import TextPairClassificationLoader
+from explainaboard.loaders.text_to_sql import TextToSQLLoader
```
Comment (odashi) [inline explainaboard/loaders/loader_factory.py:34]: I think this always requires additional installation. I think we could no longer separate this loader from others, and the following installation flag needs to be integrated into the default.

https://github.com/neulab/ExplainaBoard/blob/0a9b3bcefefcb96751fa3d663d24fff2df9dd250/setup.cfg#L39

Comment (shuaichenchang) [reply explainaboard/loaders/loader_factory.py:34]: Thank you for the review! Do you want me to create a separate PR to merge the required installations into default?

Comment (odashi) [reply explainaboard/loaders/loader_factory.py:34]: No, it would be reasonable to mix it into this request I think.

Comment (shuaichenchang) [reply explainaboard/loaders/loader_factory.py:34]: Mixed it here. Thank you!

**Review by neubig — APPROVED** [review]

**Review by odashi — APPROVED** [review]

**Pull request merged**